pydantic/_internal/_docs_extraction.py

```diff
@@ -0,0 +1,62 @@
+"""Utilities related to attribute docstring extraction."""
+from __future__ import annotations
+
+import ast
+import inspect
+import textwrap
+from typing import Any, Iterable
+
+
+def _get_assign_targets(node: ast.Assign | ast.AnnAssign) -> Iterable[str]:
+    if isinstance(node, ast.Assign):
+        for target in node.targets:
+            if isinstance(target, ast.Tuple):
+                yield from (el.id for el in target.elts)
+            else:
+                yield target.id
+    else:
+        # `target` can only be `ast.Name` in the context of a `BaseModel` subclass definition
+        yield node.target.id
+
+
+def _extract_docs(cls_def: ast.ClassDef) -> dict[str, str]:
+
+    nodes_docs_pairs = [
+        (node, inspect.cleandoc(next_node.value.s))
+        for node, next_node in zip(cls_def.body, cls_def.body[1:])
+        if isinstance(node, (ast.Assign, ast.AnnAssign))  # e.g. a: int / a = 1...
```

Comment:
We could simplify even more by disregarding `ast.Assign`, which isn't possible in a Pydantic model:

```python
class A(BaseModel):
  a = 1
# throws PydanticUserError
```

Reply:
If this is also true for dataclasses, I would be ok with taking the simpler route.

Reply:
dataclasses can have simple assignments (same goes for Pydantic dataclasses). But they aren't taken into account:

```python
from pydantic.dataclasses import dataclass

@dataclass
class A:
  a: int = 1
  b = 2

A.__pydantic_fields__
#> {'a': FieldInfo(annotation=int, required=False, default=1, init_var=False, kw_only=False)}
# Same goes for A.__dataclass_fields__, which is used to build __pydantic_fields__
```